account_move_batch_validate/models/account_move.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2014 Camptocamp SA, 2017 ACSONE
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, fields, models, _
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from odoo.addons.queue_job.job import job, Job
+except ImportError:
+    _logger.debug('Can not `import queue_job`.')
+    import functools
+
+    def empty_decorator_factory(*argv, **kwargs):
+        return functools.partial
+    job = empty_decorator_factory
+
+
+BLOCK_SIZE = 1000
+
+
+class AccountMove(models.Model):
+
+    _inherit = 'account.move'
+
+    to_post = fields.Boolean(
+        string="Posting requested", readonly=True,
+        help="Check this box to mark the move for batch posting")
+    post_job_uuid = fields.Char(string="UUID of the Job to approve this move")
+
+    @api.multi
+    @job(default_channel='root.account_move_batch_validate')
+    def validate_one_move(self):
+        if self.exists():
```

Comment:
To be correct, either this method should call `ensure_one()` or get the result of `self.exists()` and call `post()` on it. Right now we could call this method on a len(recordset) > 1 with a record filtered by `exists()`.